WIP: Add react selection-tabs component

LMS007 opened the pull request:
Fixes  #1208

LMS007 commented:
This does not include the conversion work for the `new-note-btn` component. That too, needs to be converted to react. I have more or less commented out that button in this diff. I will perform that work in a new diff and either check in before this PR or after depending on suggestions.

lyzadanger commented on src/sidebar/components/selection-tabs.js:
```diff
@@ -1,49 +1,174 @@
+    store.selectTab(type);
+  };
+
+  const showAnnotationsUnavailableMessage = function() {
```
This and the two following functions could probably be converted to basic assignment or IIFEs at the least.

lyzadanger commented on src/sidebar/components/selection-tabs.js:
```diff
@@ -1,49 +1,174 @@
+          className={classnames('selection-tabs__type', {
+            'is-selected': selectedTab === uiConstants.TAB_ANNOTATIONS,
+          })}
+          href="#"
```
I think you can remove this attr.

lyzadanger commented on src/sidebar/components/selection-tabs.js:
```diff
@@ -1,49 +1,174 @@
+          onTouchStart={selectTab.bind(this, uiConstants.TAB_ANNOTATIONS)}
+        >
+          Annotations
+          {totalAnnotations > 0 && !isWaitingToAnchorAnnotations && (
```
Might take a look to see if there is a slightly more scannable way to present this logic.

LMS007 replied:
Might be best to fold t his into a refactor effort
Best I can see it making a Tab component but since it does not have really any special logic other than passing down props, may not be super helpful.

lyzadanger commented on src/sidebar/components/selection-tabs.js:
```diff
@@ -1,49 +1,174 @@
+          className={classnames('selection-tabs__type', {
+            'is-selected': selectedTab === uiConstants.TAB_NOTES,
+          })}
+          href="#"
```
href can be removed, again.

lyzadanger commented on src/sidebar/components/selection-tabs.js:
```diff
@@ -1,49 +1,174 @@
+          onTouchStart={selectTab.bind(this, uiConstants.TAB_NOTES)}
+        >
+          Page Notes
+          {totalNotes > 0 && !isWaitingToAnchorAnnotations && (
```
Same thing here as note about scan-ability above.

lyzadanger commented on src/sidebar/components/selection-tabs.js:
```diff
@@ -1,49 +1,174 @@
+              'is-selected': selectedTab === uiConstants.TAB_ORPHANS,
+            }
+          )}
+          href="#"
```
href

lyzadanger commented on src/sidebar/components/selection-tabs.js:
```diff
@@ -1,49 +1,174 @@
+      {!isLoading() && (
+        <div className="selection-tabs__empty-message">
+          {showNotesUnavailableMessage() && (
+            <div className="annotation-unavailable-message">
```
Check these CSS classnames...unless it's not feasible, they should be prefixed with the component base class (`selection-tabs`).

LMS007 replied:
Its also used on the sidebar-content component, looks like just shared styling with different html content inside.

lyzadanger commented on src/sidebar/components/selection-tabs.js:
```diff
@@ -1,49 +1,174 @@
+        <div className="selection-tabs__empty-message">
+          {showNotesUnavailableMessage() && (
+            <div className="annotation-unavailable-message">
+              <p className="annotation-unavailable-message__label">
```
Is this `p` element truly necessary? The answer may be yes. Some of the markup here feels fiddly.

lyzadanger commented on src/sidebar/components/selection-tabs.js:
```diff
@@ -1,49 +1,174 @@
+              <p className="annotation-unavailable-message__label">
+                There are no page notes in this group.
+                <br />
+                {!settings.enableExperimentalNewNoteButton &&
```
I'd like to flag this to see if it needs to be here at all.

LMS007 commented:
- Turns out those blanks {' '} are injected by `format` anyway, so we can't wrap in <span> with spaces since it will get transformed. I noticed this in other parts of the client too.

- Broke out the tabs into a subcomponent called Tab

- Noticed there was  bug : When the value of isLoaded() changed, it did not update props and re-render the component because it was simply a function, not and not a value. I changed this to the value itself so it would trigger a re-render. 

- Added a few more tests that I felt like were not covered related to the `enableExperimentalNewNoteButton` and `isLoading`
